static/src/javascripts/projects/common/modules/ui/sticky-social.js

```diff
@@ -75,21 +75,21 @@ define([
     }
 
     function init() {
-        var testVariant = ab.getTestVariant('ShareButtons'),
+        var testVariant = ab.getTestVariant('ShareButtons2'),
             referrer,
-            socialReferrer;
+            socialContext;
 
         if (testVariant.indexOf('referrer') > -1) {
-            referrer = ((window.location.hash + '').match(/referrer=([^&]+)/) || [])[1] || document.referrer,
+            referrer = ((window.location.hash + '').match(/referrer=([^&]+)/) || [])[1] || document.referrer || '',
 
-            socialReferrer = referrer ? [
-                'facebook',
-                'twitter'
+            socialContext = [
+                {id: 'facebook', matchReferrer: 'facebook.com', matchUserAgent: 'FBAN/'},
+                {id: 'twitter', matchReferrer: 't.co', matchUserAgent: 'Twitter for iPhone'}
```

Comment:
Seems the Android Twitter app just reports the Chrome UA string - so there's nothing to differentiate it.